cdap-data-fabric/src/test/java/co/cask/cdap/data2/dataset2/lib/FileSetTest.java

```diff
@@ -227,6 +234,24 @@ public void testExternalNonExistentPath() throws IOException, DatasetManagementE
                                      .build());
   }
 
+
+  @Test(expected = IOException.class)
```

Comment:
that test case would also succeed if the first createInstance() throws that exception. Better make sure the first one succeeded and the second one failed.